### PR TITLE
feat: add a new json rpc method `get_block_economic_state`

### PR DESCRIPTION
**Pull request by yangby-cryptape**
### Purpose

Replace the JSON-RPC method [`get_cellbase_output_capacity_details`].

##### Secondary Issuance

According to [CKB RFC-0015](https://github.com/nervosnetwork/rfcs/blob/dda4478/rfcs/0015-ckb-cryptoeconomics/0015-ckb-cryptoeconomics.md#collecting-state-rent-with-secondary-issuance-and-the-nervosdao):
> Then 60% of the secondary issuance goes to the miners, 35% of the issuance goes to the NervosDAO to be distributed to the locked tokens proportionally.

Only 60% of the secondary issuance goes to the miners.
But we can't see it from the response of [`get_cellbase_output_capacity_details`].

##### Transaction Fees

According to [CKB v0.25.2 Code](https://github.com/nervosnetwork/ckb/blob/dda4ed9/util/reward-calculator/src/lib.rs#L87-L215):

```rust
    /// Miner get (tx_fee - 40% of tx fee) for tx commitment.
    /// Be careful of the rounding, tx_fee - 40% of tx fee is different from 60% of tx fee.
    pub fn txs_fees(&self, target: &HeaderView) -> Result<Capacity, Error> {
        /* ... omitted ... */
    }

    /// Earliest proposer get 40% of tx fee as reward when tx committed
    ///  ... omitted ...
    pub fn proposal_reward(&self, parent: &HeaderView, target: &HeaderView) -> Result<Capacity, Error> {
        /* ... omitted ... */
    }
```

The transaction fees which paid to miner is not same as the total transaction fee of all transactions in the mined block.
The `tx_fee` in [`get_cellbase_output_capacity_details`] is only about 60% of total transaction fee in the mined block.
And transaction fees in miner reward have two part: committed fee/reward and proposal fee/reward.

#### Help for Doing Statistics

- Currently, there is no interface can query total issuance of a block.
- Currently, there is no interface can query total transaction fee of a block.

#### Simplify the Usage

Currently, if we want to know which miner mined block-N and how many rewards he received for block-N, **we have to use a magic number 11 (it's a bit obscure, and reading the source code is the only way to get it)** and call JSON-RPC 4 times:
```rust
/* pseudo-code */
let miner =  rpc
    .get_block_hash(N)
    .map(|hash| rpc.get_block(hash).cellbase().witness().parse_miner())
    .unwrap();
let reward = rpc
    .get_block_hash(N+11)
    .map(|hash| rpc.get_cellbase_output_capacity_details(hash))
    .unwrap();
```

The new JSON-RPC method will hide the magic number and return the block hash which finalized the miner reward:
```rust
/* pseudo-code */
let hash =  rpc.get_block_hash(N).unwrap();
let miner = rpc.get_block(hash).cellbase().witness().parse_miner()).unwrap();
let reward = rpc.get_block_economic_state(hash).unwrap();
```

### Further

Currently, the `Transaction Fee` on the ckb explorer is the sum of committed reward and proposal reward in miner reward of a block, not the total transaction fee of all transactions in that block.
**It's really hard to understand the `Transaction Fee` for newbies.**

For example:
- [block-14 on mainnet](https://explorer.nervos.org/block/0x5b78979ee71a7a32868e6256874571b2b59a9e40e38e6923e653db4ae5fd3d1d) has no proposal transaction, and total transaction fee of it is `66.00000000 CKB`, but the `Transaction Fee` on the ckb explorer is `39.60000000 CKB`.
- [block-334 on mainnet](https://explorer.nervos.org/block/0xbaf6c624282a7a9c5bc673ea79d6b612de822f68cb8e3f8004bf887dd11583de) has no transaction expect cellbase, but the `Transaction Fee` on the ckb explorer is `0.40522815 CKB`.

**I think we should unify the names.**
And, I didn't find any description about how to partition the total transaction fee of all transactions in a block into committed fee/reward and proposal fee/reward in [CKB RFCs](https://github.com/nervosnetwork/rfcs/tree/dda4478/rfcs).
Did I miss it?

[`get_cellbase_output_capacity_details`]: https://github.com/nervosnetwork/ckb/tree/dda4ed9/rpc#get_cellbase_output_capacity_details

### CODE REVIEW

File: rpc/json/rpc.json [hunk 1]
```diff
@@ -751,6 +751,33 @@
...
+                "proposal": "0x0",
+                "secondary": "0x17b93605"
+            },
+            "txs_fee": "0x0"
+        },
+        "types": [
+            {
+                "hash": "Block hash"
+            }
+        ]
```
Comment (quake) [inline rpc/json/rpc.json]: could you add a `returns` section to descript the result, reference:
https://github.com/nervosnetwork/ckb/blob/develop/rpc/json/rpc.json#L242

Comment (yangby-cryptape) [reply rpc/json/rpc.json]: Yes! I will do it later.
I'm not anxious for merging this PR.
**I'm waiting for more opinions.**

File: rpc/README.md [hunk 2]
```diff
@@ -650,6 +651,51 @@ http://localhost:8114
 }
 ```
 
+### `get_block_economic_state`
+
+Returns increased issuance, miner reward and total transaction fee of a block.
```
Comment (doitian) [inline rpc/README.md]: What's the behavior when the proposal rewards are not finalized yet?

File: rpc/json/rpc.json [hunk 3]
```diff
@@ -751,6 +751,33 @@
...
+                "primary": "0x18ce922bca",
+                "secondary": "0x7f02ec655"
+            },
+            "miner_reward": {
+                "committed": "0x0",
+                "primary": "0x18ce922bca",
+                "proposal": "0x0",
+                "secondary": "0x17b93605"
+            },
+            "txs_fee": "0x0"
```
Comment (doitian) [inline rpc/json/rpc.json]: There's another field we may want to include in the response, total withdrawn DAO rewards of all the txs.

File: rpc/README.md (hunk 2 above)
Comment (yangby-cryptape) [reply rpc/README.md]: [Return `None` if not finalized](https://github.com/nervosnetwork/ckb/pull/1848/files#diff-c80a342630d6634092de83806d174b3eR306-R309), same as [`get_cellbase_output_capacity_details`](https://github.com/nervosnetwork/ckb/tree/dda4ed9/rpc#get_cellbase_output_capacity_details).

File: rpc/json/rpc.json (hunk 1 above)
Comment (yangby-cryptape) [reply rpc/json/rpc.json]: [DONE!](https://github.com/nervosnetwork/ckb/pull/1848/commits/deb5d3f4ac3d37edc9a615723efe6b35dc7baa5c)

**Comment by doitian** [comment]
bors r=quake,doitian

**Comment by doitian** [comment]
bors ping

**Comment by doitian** [comment]
bors try

**Comment by doitian** [comment]
bors r=quake,doitian

**Pull request merged**